packages/@aws-cdk-testing/cli-integ/lib/process.ts

```diff
@@ -0,0 +1,139 @@
+import * as child from 'child_process';
+import * as pty from 'node-pty';
+
+/**
+ * IProcess provides an interface to work with a subprocess.
+ */
+export interface IProcess {
+
+  /**
+   * Register a callback to be invoked when a chunk is written to stdout.
+   */
+  onStdout(callback: (chunk: Buffer) => void): void;
+
+  /**
+   * Register a callback to be invoked when a chunk is written to stderr.
+   */
+  onStderr(callback: (chunk: Buffer) => void): void;
+
+  /**
+   * Register a callback to be invoked when the process exists.
+   */
+  onExit(callback: (exitCode: number) => void): void;
+
+  /**
+   * Register a callback to be invoked if the process failed to start.
+   */
+  onError(callback: (error: Error) => void): void;
+
+  /**
+   * Write the process stdin stream.
+   */
+  writeStdin(data: string): void;
+
+};
+
+export interface ProcessSpawnOptions {
+  /**
+   * Start a TTY enabled process. Should be used when starting an interactive process.
+   *
+   * @default false
+   */
+  readonly tty?: boolean;
+
+  /**
+   * Allow any option to pass through to the underlying implementation.
+   *
+   * It's tricky to create a common interface because options can vary quite significantly
+   * between a tty and a non tty process.
+   */
+  [key: string]: any;
```

Comment:
In that case:

```ts
type ProcessSpawnOptions = 
  | { tty: true } & pty.ProcessOptions
  | { tty?: false } & nodejs.ProcessSpawnOptions
  ;
```

Or something to that effect?

Reply:
Its kind of messy because of overlapping keys with different types. I split into two static function to make `Process` a bit more type safe, coercing the options one level up.